cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -94,52 +89,9 @@ func PerformPostUpgradeTasks(client clientset.Interface, cfg *kubeadmapi.InitCon
 		errs = append(errs, err)
 	}
 
-	if err := dns.EnsureDNSAddon(&cfg.ClusterConfiguration, client); err != nil {
```

Comment:
the way we approached this for other commands such as init or join was the following:
- given one big function that does the work instead of phases
- create a phase runner for a command (in this case "apply")
- run the runner by default
- start breaking out logic from the main function into phases that the runner runs
- after the runner is done call the big function with remaining logic that is not in phases.

with addons being last in a function like PerformPostUpgradeTasks, not sure how this breaking up of the addons phase will work by preserving the upgrade to work.

i think the breakup into phases should be done down->up (addons last).

Reply:
All this was was to remove it from postupgrade, it did not make it a phase. originally if you had tried to make `PerformPostUpgradeTasks` a phase you would have had to do more then just make it's funcation a phase, you would have to also remove `Addons` from it. 

My goal was at the end of these have them all in a line in the one big function `step one` in your analogy above. All this did was break the logic up so it could be made it's own phase when needed. That is all any of this was/is. Breaking logical parts out so they can be made into the phases we talked about.

All that being said. if people want addons to be left inside another phase. that is fine with me. :) I'll leave them there! :+1: